framework/src/org/checkerframework/framework/test/DefaultCheckerTest.java

```diff
@@ -0,0 +1,81 @@
+package org.checkerframework.framework.test;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.annotation.processing.AbstractProcessor;
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.checkerframework.framework.test.TestConfigurationBuilder.buildDefaultConfiguration;
+
+/**
+ * To use this class you must do two things:
+ * 1) Create exactly 1 constructor in the subclass with exactly 1 argument which is a java.io.File.  This File
+ * will be the Java file that is compiled and whose output is verified.
+ * 2) Create a public static method that is annotation with org.junit.runners.Parameterized.Parameters.  This method
+ * should look like:
+ * {@code, @Parameters List<Object[]> getTestFiles(); }
+ * The return type should be a List of 1 element arrays, where each member of that array is a Java file to be tested.
```

Comment:
Well I guess it should be:
I can update this comment. 
Alternatively:  I kept this signature because it matches the original framework.  We could make this a List<String> and do the conversion to a List<Object []> in the TestSuite to make this less ambiguous.    It needs to be List<Object []> because that's what JUnit expects for parameters.